packages/wp-source/src/libraries/handlers/date.ts

```diff
@@ -43,20 +44,42 @@ export const dateHandler: Handler = async ({
     throw new ServerError(`date "${path}" doesn't have page ${page}`, 404);
 
   // 4. get posts and pages count
-  const total = getTotal(response);
-  const totalPages = getTotalPages(response);
+  const total = getTotal(response, items.length);
+  const totalPages = getTotalPages(response, 0);
+
+  // returns true if next page exists
+  const hasOlderPosts = page < totalPages;
+  // returns true if previous page exists
+  const hasNewerPosts = page > 1;
+
+  const getPageLink = (page: number) =>
+    libraries.source.stringify({ path, query, page });
 
   // 5. add data to source
-  Object.assign(state.source.data[route], {
+  const currentPageData = state.source.data[route];
+
+  const newPageData: DateData = {
     year,
     month,
     day,
     items,
     total,
     totalPages,
     isArchive: true,
-    isDate: true
-  });
+    isDate: true,
+    isFetching: currentPageData.isFetching,
+    isReady: currentPageData.isReady,
+    prev: hasOlderPosts ? getPageLink(page - 1) : undefined,
+    next: hasNewerPosts ? getPageLink(page + 1) : undefined
+  };
+
+  Object.assign(currentPageData, newPageData);
```

Comment:
not super happy that this is basically repeated 4 times in `taxonomy`, `data`, `author` and `archivePostType` handlers, but extracting it to a function is bad too because we depend on the state from:
```
 const { api, populate, parse, getTotal, getTotalPages } = libraries.source;
 const { path, page, query } = parse(route);
```
so we'd end up passing a lot of function arguments around